test/UnitTests/CheckoutControllerTest.cs

```diff
@@ -0,0 +1,168 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Threading;
+using Microsoft.AspNet.Http.Core;
+using Microsoft.AspNet.Http.Core.Collections;
+using Microsoft.AspNet.Mvc;
+using Microsoft.AspNet.Routing;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.DependencyInjection.Fallback;
+using Xunit;
+using MusicStore.Controllers;
+using MusicStore.Models;
+
+namespace UnitTests
+{
+    public class CheckoutControllerTest
+    {
+        IServiceProvider _serviceProvider;
+
+        public CheckoutControllerTest()
+        {
+            var services = new ServiceCollection();
+
+            services.AddEntityFramework()
+                      .AddInMemoryStore()
+                      .AddDbContext<MusicStoreContext>();
+
+            _serviceProvider = services.BuildServiceProvider();
+        }
+
+        [Fact]
+        public void AddressAndPaymente_ReturnsDefaultView()
+        {
+            // Arrange
+            var controller = new CheckoutController();
+
+            // Act
+            var result = controller.AddressAndPayment();
+
+            // Assert
+            var viewResult = Assert.IsType<ViewResult>(result);
+            Assert.Null(viewResult.ViewName);
+        }
+
+        [Fact]
+        public void AddressAndPayment_ReturnsOrderIfInvalidPromoCode()
+        {
+            // Arrange
+            var context = new DefaultHttpContext();
+            context.Request.Form =
+                new FormCollection(new Dictionary<string, string[]>());
```

Comment:
@rynowak 
that's correct, i missed the test, and it will be the same test to check the DB status change from order. But..that code path uses ShoppingCart, which uses Session. I didn't include that test in this PR. I'll make sure to add that test after figuring out how to handle Session.